### ISSUE TITLE
Expose 'undo at exit' to runtime

### ISSUE THREAD

**Issue by BluePrintRandom** [UPBGE/upbge#1106]
I don't know if there already is a binding for it,
but I need to be able to set 'undo at exit' in game.

something like

bge.logic.undoAtExit = bool

**Comment by marechal-p** [comment]
What does "undo at exit"?

**Comment by BluePrintRandom** [comment]
it undoes all changes to a scene to like before  pressing P in the embedded player.

**Comment by BluePrintRandom** [comment]
I need to be able to save or abandon changes from inside the running engine basically - it's to make a custom python tool interface for bpy meshes.

**Comment by youle31** [comment]
There should already be a bpy API for that. Maybe search for undo_push or just undo in blender API.

When we do a simple usage of bge without using bpy, normally the changes done to objects (like object position) are automatically reverted/undone at bge exit.

But when we begin to use bpy to deform meshes, edit materials, sculpt, paint, play with modifiers... the changes are reverted/undone only if "undo at exit option" is checked.
Basically it saves (undo_push) the .blend state at bge start, and restores it at bge exit. The code is in InitEngine/ExitEngine iirc.

https://docs.blender.org/api/master/search.html?q=undo&check_keywords=yes&area=default

**Comment by BluePrintRandom** [comment]
do you think you could provide a sample of py for this?

I don't get it.
